packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -326,7 +360,7 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
 
       // Insert a dummy witness if no witness exists
       if (typeof witnessIndex !== 'number') {
-        witnessIndex = this.createWitness();
+        witnessIndex = this.addEmptyWitness();
```

Comment:
What condition would we want a dummy witness? Will these be used in production?

Reply:
A dummy witness is used for gas estimation. The correct witness needs to added at the end of the tx preparation flow as it hashes the entire transaction. So when we are just estimating, we just need to make sure we have added 32 bytes to get the correct estimation.